analyses/snv-callers/scripts/00-set_up.R

```diff
@@ -0,0 +1,144 @@
+# is used here.
+cosmic_file <- file.path(snv_dir, "CosmicMutantExport.tsv.gz")
+
+# This is the cleaned version that only contains the genomic coordinates and
+# base changes from the original file for brain sample mutations only.
+cosmic_clean_file <- file.path(cosmic_dir, "brain_cosmic_variants_coordinates.tsv")
+
+# Will set up this annotation file below
+annot_rds <- file.path(scratch_dir, "hg38_genomic_region_annotation.rds")
+
+# This will be used for all WXS samples, but WGS bed regions are caller-specific
+wxs_bed_file <- file.path(data_dir, "WXS.hg38.100bp_padded.bed")
+
+################################ Build Annotation ##############################
+# We will only run this if it hasn't been run before
+if (!file.exists(annot_rds)) {
+  # Print progress message
+  message("Setting up genomic region annotation file. Only need to do this once.")
+
+  # Here we are specifying what types of annotation we would like
+  annots <- c(
+    "hg38_basicgenes",
+    "hg38_genes_intergenic",
+    "hg38_genes_intronexonboundaries",
+    "hg38_genes_exonintronboundaries",
+    "hg38_genes_5UTRs",
+    "hg38_genes_exons",
+    "hg38_genes_introns",
+    "hg38_genes_3UTRs"
+  )
+
+  # Build all the annotations into GRanges object
+  annotations <- annotatr::build_annotations(genome = "hg38", annotations = annots)
+
+  # Write this object so we don't have to write it again
+  readr::write_rds(annotations,
+    annot_rds,
+    compress = "gz"
+  )
+}
+######################## Set up COSMIC Mutations file ###########################
+# This set up will not be run unless you obtain the original data file
+# from https://cancer.sanger.ac.uk/cosmic/download , these data are available
+# if you register. The full, unfiltered somatic mutations file
+# CosmicMutantExport.tsv for grch38 is used here but only mutations from brain
+# related samples are kept.
+if (!file.exists(cosmic_clean_file)) {
+  # Print progress message
+  message("Setting up COSMIC mutation file. Only need to do this once.")
+
+  # Read in original file
+  cosmic_variants <- data.table::fread(cosmic_file,
+    data.table = FALSE
+  ) %>%
+    # Keep only brain mutations so the file is smaller
+    dplyr::filter(`Site subtype 1` == "brain")
+  # Get rid of spaces in column names
+  cosmic_variants <- cosmic_variants %>%
+    dplyr::rename_all(dplyr::funs(stringr::str_replace_all(., " ", "_"))) %>%
+    # Separate the genome coordinates into their own BED like variables
+    dplyr::mutate(
+      Chromosome = paste0(
+        "chr",
+        stringr::word(Mutation_genome_position, sep = ":", 1)
+      ),
+      Start_Position = stringr::word(Mutation_genome_position, sep = ":|-", 1),
+      End_Position = stringr::word(Mutation_genome_position, sep = "-", 2),
+      # Make a base_change variable so we can compare to our set up for PBTA data
+      base_change = substr(Mutation_CDS, nchar(Mutation_CDS) - 2, 10)
```

Comment:
I'm a bit confused here. Why the chosen start and stop positions? It looks like you want the last 3 characters, but if the string is longer than 10, you will not get that?

Reply:
The strings are showing the base changes which are all 3 characters long. They don't have the insertions or deletions spelled out so all of them are `ins` `del` or `G>T`-like. I could change it to use `nchar(Mutation_CDS)` if that is less confusing.

Reply:
I went ahead and changed it to `nchar(Mutation_CDS)` because I don't think you are the first person to be confused by that.